hdfs/client.py

```diff
@@ -602,6 +602,24 @@ def rename(self, hdfs_src_path, hdfs_dst_path):
         self.resolve(hdfs_src_path), hdfs_dst_path
       )
 
+  def make_directory(self, hdfs_path, permissions=None):
+    """Make a directory, returning True if successful, or False if a file or
+    directory already exists at the path.
+
+    :param hdfs_path: Path of directory to be made
+    :param permissions: Optional, octal permissions string.
+
+    """
+    try:
+      self.status(hdfs_path):
```

Comment:
We shouldn't be making an extra request for for this, just using the boolean returned by the API (that's what I meant by "if possible", sorry for not being more explicit). Also, this will currently return `False` if the path exists but isn't a directory (i.e. not good behavior).